src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputModelTypeConverter.cs

```diff
@@ -132,6 +155,34 @@ private static void CreateProperties(ref Utf8JsonReader reader, ICollection<Inpu
             while (reader.TokenType != JsonTokenType.EndArray)
             {
                 var property = reader.ReadWithConverter<InputModelProperty>(options);
+                if (property != null && isMultipartType)
+                {
+                    var propertyType = property.Type;
+                    switch (property.Type)
+                    {
+                        case InputPrimitiveType primitiveType when primitiveType.Name.Equals("Bytes"):
+                            propertyType = InputPrimitiveType.Stream;
+                            break;
+                        case InputListType listType when listType.ElementType.Name.Equals("Bytes"):
+                            propertyType = new InputListType(listType.Name, InputPrimitiveType.Stream, listType.IsEmbeddingsVector, listType.IsNullable);
+                            break;
+                        case InputDictionaryType dictionaryType when dictionaryType.ValueType.Name.Equals("Bytes"):
+                            propertyType = new InputDictionaryType(dictionaryType.Name, dictionaryType.KeyType, InputPrimitiveType.Stream, dictionaryType.IsNullable);
+                            break;
+                        default:
+                            break;
+                    }
+
+                    property = new InputModelProperty(property.Name,
+                                                    property.SerializedName,
+                                                    property.Description,
+                                                    propertyType,
+                                                    property.ConstantValue,
+                                                    property.IsRequired,
+                                                    property.IsReadOnly,
+                                                    property.IsDiscriminator,
+                                                    property.FlattenedNames);
+                }
```

Comment:
this looks so weird that we are changing the serialization result, which we should not.
We should do this when building the property of the generated model in `ModelTypeProviderFields`

Reply:
I will try.

Reply:
@ArcturusZhang  The InputType will used when serialize it. So we need to update here. It is a workaround, when we adopt tcgc, we will do this transformation in emitter. Current we will keep to change here.

Reply:
I think you did not get my point here.
My point is that we should do this type transformation when we determine the output type of the property.